homeassistant/components/alexa/smart_home.py

```diff
@@ -383,12 +377,84 @@ def get_property(self, name):
             raise _UnsupportedProperty(name)
 
         unit = self.entity.attributes[CONF_UNIT_OF_MEASUREMENT]
+        temp = self.entity.state
+        if self.entity.domain == climate.DOMAIN:
+            temp = self.entity.attributes.get(
+                climate.ATTR_CURRENT_TEMPERATURE, 0)
         return {
-            'value': float(self.entity.state),
+            'value': float(temp),
             'scale': API_TEMP_UNITS[unit],
         }
 
 
+class _AlexaThermostatController(_AlexaInterface):
+    def name(self):
+        return 'Alexa.ThermostatController'
+
+    def properties_supported(self):
+        properties = []
+        supported = self.entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE:
+            properties.append({'name': 'targetSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_LOW:
+            properties.append({'name': 'lowerSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_HIGH:
+            properties.append({'name': 'upperSetpoint'})
+        if supported & climate.SUPPORT_OPERATION_MODE:
+            properties.append({'name': 'thermostatMode'})
+        return properties
+
+    def properties_retrievable(self):
+        return True
+
+    def get_property(self, name):
+        if name == 'thermostatMode':
+            mode = self.entity.state.upper().replace('IDLE', 'OFF')
+            if mode not in CLIMATE_SUPPORTED_MODES:
+                return {
+                    'value': 'CUSTOM',
+                    'customName': mode,
+                }
+            return mode
+
+        temp = 0
+        unit = self.entity.attributes[CONF_UNIT_OF_MEASUREMENT]
+        if name == 'targetSetpoint':
+            temp = self.entity.attributes.get(ATTR_TEMPERATURE, 0)
+        elif name == 'lowerSetpoint':
+            temp = self.entity.attributes.get(climate.ATTR_TARGET_TEMP_LOW, 0)
+        elif name == 'upperSetpoint':
+            temp = self.entity.attributes.get(climate.ATTR_TARGET_TEMP_HIGH, 0)
```

Comment:
Drop these defaults. This is definitely wrong.

It is very important that our implementation is very strict. We follow the rules as set by both Alexa and Home Assistant (like only supporting official climate modes). We should not make up values to try to satisfy Alexa, in that case we should prevent the entity from syncing so that people can fix the integrations.